### PR TITLE
[GPU] Remove DRAM size from model_str in device description.

### PR DESCRIPTION
**Pull request by sergachev**
Exact DRAM size can slightly vary for the same GPU model so it shouldn't be used in model identification. The removed part of gpu_compiler_test_autotune_db.textproto illustrates that.

### CODE REVIEW

**Review by derdrdirk — APPROVED** [review]

**Comment by dimitar-asenov** [comment]
I just saw this PR. We're aware of the problem and I'm actually working on a different solution that also excludes the RAM, but also makes other changes to the cache string. It should be ready today.

**Comment by dimitar-asenov** [comment]
FYI: the alternative PR is ready, but we need to wait until Thursday to submit it. Conceptually, it's very similar to this one.

**Comment by dimitar-asenov** [comment]
Here is the current version of the other PR: https://github.com/openxla/xla/pull/14954

> Then what's the reason to prefer it?

The other PR doesn't change the model_str in cuda executor. It touches only the autotuning key and does a cleanup related to deviceless autotuning keys. It also does changes that are needed internally (and not visible on GitHub). At the point I became aware of your PR we already had the internal one nearly done. It's now fully ready but we have to wait for an extra internal approval until tomorrow.

**Pull request closed without merging**
